go/vt/sqlparser/sql.y

```diff
@@ -2331,6 +2370,7 @@ reserved_keyword:
 | LOCALTIMESTAMP
 | LOCK
 | MATCH
+| MAXVALUE
```

Comment:
There are docs on `reserved_keyword` and `non_reserved_keyword` -- those should be used over that keywords.html link.  If the docs are not clear enough feel free to fix as part of this PR.

Our reserved_keyword and non_reserved_keyword are not meant to mirror mysql -- it's for the sake of conflicts in our grammar. In many cases that may be in line with mysql, since we have some similar constructs. But ours is also a much simpler grammar and so less things should be reserved.